discretize/BaseMesh.py

```diff
@@ -44,7 +60,7 @@ def dim(self):
         :rtype: int
         :return: dim
         """
-        return self._dim
+        return len(self.n)
```

Comment:
For all of these, actually, possibly worth checking if things (`self.n`, `self.dims`, etc.) are `None`.